[Toolkit] Improve descriptions of Shadcn components (#2728)
symfony/ux: src/Toolkit/kits/shadcn/docs/components/Table.md

# Table

> Displays data in a structured grid format with rows and columns, supporting headers, captions, and footers.

Reviewer: This one is the only one starting with "Displays" here is a suggestion to match the other's format :) 
```suggestion
A structured grid element that organizes data into rows and columns, supporting headers, captions, and footers.
```
